src/Marten/Util/SharedBuffer.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Buffers;
+using System.IO;
+
+namespace Marten.Util
+{
+    internal readonly struct SharedBuffer : IDisposable
```

Comment:
**FIX:** @solyutor, looking at the failing tests, you're missing skipping SOH control character (https://github.com/JasperFx/marten/blob/b5325056d6226159565036b0eda89d0b89ff5429/src/Marten/Util/StreamExtensions.cs#L15). 
See [FromJsonAsync](https://github.com/JasperFx/marten/pull/1967/files#diff-860d4464563fb983fc220a5da7bcf182e25876de07eccf88e720e2fb1c578517R101) implementation.

Reply:
Thanks for pointing to the problem. I also missed `_deserializeOptions` and fixed this as well.